packages/@aws-cdk/aws-redshift/lib/database-query.ts

```diff
@@ -0,0 +1,103 @@
+import * as path from 'path';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import * as cdk from '@aws-cdk/core';
+import * as customresources from '@aws-cdk/custom-resources';
+import { Construct } from 'constructs';
+import { Cluster } from './cluster';
+import { DatabaseProps } from './database-props';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+export interface DatabaseQueryProps extends DatabaseProps {
+  readonly handler: string;
+  readonly properties?: {[key: string]: any};
+  /**
+   * The policy to apply when this resource is removed from the application.
+   *
+   * @default cdk.RemovalPolicy.Destroy
+   */
+  readonly removalPolicy?: cdk.RemovalPolicy;
+}
+
+export class DatabaseQuery extends CoreConstruct implements iam.IGrantable {
+  readonly grantPrincipal: iam.IPrincipal;
+  readonly ref: string;
+
+  private readonly resource: cdk.CustomResource;
+
+  constructor(scope: Construct, id: string, props: DatabaseQueryProps) {
+    super(scope, id);
+
+    const adminUser = this.getAdminUser(props);
+    const handler = new lambda.SingletonFunction(this, 'Handler', {
+      code: lambda.Code.fromAsset(path.join(__dirname, 'database-query-provider')),
+      runtime: lambda.Runtime.NODEJS_14_X,
+      handler: 'index.handler',
+      timeout: cdk.Duration.minutes(1),
```

Comment:
Seems short. Are you sure any DDL update will finish in this time?

Reply:
The queries take less than 15 seconds to execute, from what I've seen. More than happy to bump it up just in case, though. What do you think a good value would be?